ui/components/component-library/popover/popover.stories.tsx

```diff
@@ -0,0 +1,807 @@
+import React, { useState, useEffect } from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import Box from '../../ui/box/box';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  Color,
+  DISPLAY,
+  JustifyContent,
+  TEXT_ALIGN,
+} from '../../../helpers/constants/design-system';
+import { Icon, IconName, IconSize, PopoverHeader, Text } from '..';
+import README from './README.mdx';
+import { Popover, PopoverPosition, PopoverRole } from '.';
+
+export default {
+  title: 'Components/ComponentLibrary/Popover',
+  component: Popover,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  argTypes: {
+    children: {
+      control: 'text',
+    },
+    position: {
+      options: PopoverPosition,
+      control: 'select',
+    },
+    role: {
+      options: PopoverRole,
+      control: 'select',
+    },
+    className: {
+      control: 'text',
+    },
+  },
+  args: {
+    children: 'Popover',
+  },
+} as ComponentMeta<typeof Popover>;
+
+const Template: ComponentStory<typeof Popover> = (args) => {
+  const [referenceElement, setReferenceElement] = useState();
+  const [isOpen, setIsOpen] = useState(true);
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  // Example of how to use mouse events to open and close popover
+  // const handleMouseEnter = () => {
+  //   setIsOpen(true);
+  // };
+
+  // const handleMouseLeave = () => {
+  //   setIsOpen(false);
+  // };
+
+  // Example of how open popover with focus
+  // pair with onBlur (example using handleClose) to close popover
+  // const handleFocus = () => {
+  //   setIsOpen(true);
+  // };
+
+  // const handleClose = () => {
+  //   setIsOpen(false);
+  // };
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Escape') {
+      setIsOpen(false);
+    }
+  };
+
+  // Example of how to use keyboard events to close popover with escape key
+  useEffect(() => {
+    if (isOpen) {
+      document.addEventListener('keydown', handleKeyDown);
+    } else {
+      document.removeEventListener('keydown', handleKeyDown);
+    }
+  }, [isOpen]);
+
+  // Example of how to use ref to open popover
+  const setBoxRef = (ref) => {
+    setReferenceElement(ref);
+  };
+
+  return (
+    <Box
+      display={DISPLAY.FLEX}
+      justifyContent={JustifyContent.center}
+      alignItems={AlignItems.center}
+      style={{ width: '200vw', height: '200vh' }}
+    >
+      <Box
+        ref={setBoxRef}
+        onClick={handleClick}
+        // onMouseEnter={handleMouseEnter}
+        // onMouseLeave={handleMouseLeave}
+        // onFocus={handleFocus}
+        // onBlur={handleClose}
+        backgroundColor={BackgroundColor.primaryAlternative}
+        style={{ width: 200, height: 200 }}
+        color={Color.primaryInverse}
+        as="button"
+      >
+        Click to toggle popover
+      </Box>
+      <Popover
+        referenceElement={referenceElement}
+        isOpen={isOpen}
+        {...args}
+      ></Popover>
```

Comment:
You have a few tags that could use the self closing syntax

<img width="254" alt="Screenshot 2023-04-27 at 9 42 32 PM" src="https://user-images.githubusercontent.com/8112138/234858405-39b37c3e-c5da-44f5-bdaa-01ceae31df4d.png">